source/kraken/fill_disruption_from_chaos.cpp

```diff
@@ -46,6 +48,40 @@ namespace nt = navitia::type;
 namespace bt = boost::posix_time;
 namespace bg = boost::gregorian;
 
+namespace {
+
+struct DiscreteVJTag{};
+struct FrequencyVJTag{};
+
+template<typename VJ_T>
+struct VJTag_Traits{};
+
+template<>
+struct VJTag_Traits<nt::DiscreteVehicleJourney>{
+    typedef DiscreteVJTag Tag;
+};
+
+template<>
+struct VJTag_Traits<nt::FrequencyVehicleJourney>{
+    typedef FrequencyVJTag Tag;
+};
+
+template<typename T>
+using Tag_t = typename VJTag_Traits<T>::Tag;
+
+auto get_vj_list_of_jp_helper(nt::JourneyPattern* jp, DiscreteVJTag)
+-> decltype((jp->discrete_vehicle_journey_list)) { return jp->discrete_vehicle_journey_list;}
+
+auto get_vj_list_of_jp_helper(nt::JourneyPattern* jp, FrequencyVJTag)
+-> decltype((jp->frequency_vehicle_journey_list)) { return jp->frequency_vehicle_journey_list;}
+
```

Comment:
In order to simplify this chaotic function overloading, I have two solutions for that, which one do you guys prefer?

```c++
template<typename T>
std::vector<std::unique_ptr<T>>& get_vj_list_of_jp_helper(nt::JourneyPattern* jp);

template<>
auto get_vj_list_of_jp_helper<nt::DiscreteVehicleJourney>(nt::JourneyPattern* jp)
-> decltype((jp->discrete_vehicle_journey_list))
{ return jp->discrete_vehicle_journey_list; }

template<>
auto get_vj_list_of_jp_helper<nt::FrequencyVehicleJourney>(nt::JourneyPattern* jp)
-> decltype((jp->frequency_vehicle_journey_list))
{return jp->frequency_vehicle_journey_list;}
```

and second one:

```c++ 
template<typename T,
         typename = typename std::enable_if<std::is_same<T, nt::DiscreteVehicleJourney>::value>::type
        >
auto get_vj_list_of_jp_helper(nt::JourneyPattern* jp)
-> decltype((jp->discrete_vehicle_journey_list)) {
    return jp->discrete_vehicle_journey_list;
}

template<typename T,
         typename = typename std::enable_if<std::is_same<T, nt::FrequencyVehicleJourney>::value>::type
        >
auto get_vj_list_of_jp_helper(nt::JourneyPattern* jp)
-> decltype((jp->frequency_vehicle_journey_list)) {
    return jp->frequency_vehicle_journey_list;
}
```